### PR TITLE
Playas gonna play

### PR DESCRIPTION
**Pull request by kelseyedmodo**

### CODE REVIEW

**Comment by kelseyedmodo** [comment]
Requests aren't quite right. Can't figure out how to get past this 500 error.  

Players collection in right format though!

File: routes/teams.js [hunk 1]
```diff
@@ -23,4 +24,101 @@ router.get('/team', function(req, res) {
     });
 });
 
+
+// when players endpoint is hit, call the API/DB using that team_id
+router.get('/players', function(req, res) {
+  temp = req.query.split('=');
```
Comment (blakeparkinson) [inline routes/teams.js]: why is this needed?

Comment (kelseyedmodo) [inline routes/teams.js]: because the query comes back as "team_id=1234567"... and I only want the id, not the label.

File: routes/teams.js [hunk 2]
```diff
@@ -23,4 +24,101 @@ router.get('/team', function(req, res) {
...
+  team_id = temp[1];
+  console.log("team id is: "+team_id);
+  fetchPlayers(team_id)
```
Comment (blakeparkinson) [inline routes/teams.js]: it looks like you are missing the final piece to return the json response. it should be something like ```players = fetchPlayers(team_id);     	res.json(players);```

File: routes/teams.js (hunk 1 above)
Comment (blakeparkinson) [inline routes/teams.js]: i think there is something maybe deeper in that object that will parse that out already? maybe not

**Comment by blakeparkinson** [comment]
looks pretty good though. This was the hardest piece of the entire thing so far

**Comment by kelseyedmodo** [comment]
@blakeparkinson    still not getting the correct return to client. But, if you can give me a hint there, you can start using the data while I polish up the date stuff/ team_name info.

File: routes/teams.js [hunk 3]
```diff
@@ -40,4 +43,115 @@ router.get('/team', function(req, res) {
     });
 });
 
+
+// when players endpoint is hit, call the API/DB using that team_id
+router.get('/players', function(req, res) {
+  team_id = req.query["team_id"];
+  console.log("team id is: "+team_id);
+  players = fetchPlayers(team_id);
```
Comment (blakeparkinson) [inline routes/teams.js]: You are setting players to the fetchplayes method here, but fetchplayers does not return anything

File: routes/teams.js [hunk 4]
```diff
@@ -40,4 +43,115 @@ router.get('/team', function(req, res) {
...
+});
+
+
+
+
+
+  
+var fetchPlayers = function(team_id){ 
+      db_players.collection('players').find({team_id : team_id}).toArray(function (err, items){
+        secondFunction(items, team_id);
```
Comment (blakeparkinson) [inline routes/teams.js]: Assign a variable to this and return that variable

File: routes/teams.js [hunk 5]
```diff
@@ -16,12 +16,15 @@ var golf_key = 'wuczn4z2ktufacuae7u8sxfc';
...
 var express = require('express');
+var request = require('request');
 var router = express.Router();
 var http = require("http"),
     mongojs = require("mongojs"),
-    uri = 'mongodb://root:root@ds031541.mongolab.com:31541/rosterblitz';
+    uri = 'mongodb://root:root@ds031541.mongolab.com:31541/rosterblitz',
+
+    db = mongojs.connect(uri, ["teams"]),
+    db_players = mongojs.connect(uri, ["players"]);
```
Comment (blakeparkinson) [inline routes/teams.js]: We don't need two variables for this, since the collection is specified in the open function. Eliminate the 2nd param, and just with the db variable

File: routes/teams.js (hunk 4 above)
Comment (kelseyedmodo) [inline routes/teams.js]: It's not that simple. If I just assign it to a variable and return it, the return runs before the db call is complete so I return an empty item.  But I can't figure out how to make it wait for the data to return first.

File: routes/teams.js [hunk 6]
```diff
@@ -34,10 +34,107 @@ router.get('/', function(res, res) {
...
+
+var returnPlayers = function (players, res){
+    res.json(players);
+}
+
+  
+var fetchPlayers = function(team_id, res, callback){
+  db.collection('players').find({team_id : team_id}).toArray(function (err, items){
+    if (items.length > 0){ // data in Mongo
+      var date = items[0]["last_updated"].valueOf();
```
Comment (blakeparkinson) [inline routes/teams.js]: we are using underscore so you can also do ```_.first(items['last_updated']``` . I dont think the valueof is needed

File: routes/teams.js [hunk 7]
```diff
@@ -34,10 +34,107 @@ router.get('/', function(res, res) {
...
+      var datenow = new Date();
+      var datecutoff = datenow.getTime() - 86400000;
```
Comment (blakeparkinson) [inline routes/teams.js]: can you add comment for what's going on here..or why we are doing it?

File: routes/teams.js [hunk 8]
```diff
@@ -34,10 +34,107 @@ router.get('/', function(res, res) {
...
+    playersarray[i] = {};
+    for(var key in response.players[i]){
+      if (key != "total" && key != "average"){
+        var value = response.players[i][key];
+        playersarray[i][key] = value;
+      }
+      else if (key =="total"){
+        for(var key in response.players[i].total) {
+          var value = response.players[i].total[key];
+          playersarray[i]["total_"+key] = value;
```
Comment (blakeparkinson) [inline routes/teams.js]: im confused, the index will be called "total_total"

File: routes/teams.js [hunk 9]
```diff
@@ -34,10 +34,107 @@ router.get('/', function(res, res) {
...
+  return teamDocument;
+}
+
+
+function mongoInsertPlayers(team_id, team_document){
+  console.log("inserting into the DB");
+  db.open(function(err, db){
+    db.collection("players").update({team_id: team_id},
+    {$set: {team_id: team_document["team_id"], last_updated: new Date(), players: team_document["players"]}},
+    {upsert: true, multi:false}, function (err, upserted){
```
Comment (blakeparkinson) [inline routes/teams.js]: nice...mongo is pretty cool

File: routes/teams.js [hunk 10]
```diff
@@ -34,10 +34,107 @@ router.get('/', function(res, res) {
...
+      // if error
```
Comment (blakeparkinson) [inline routes/teams.js]: can you just log something out if we are in here?

**Comment by blakeparkinson** [comment]
probably gonna be quite a bit of merge conflicts with this one

File: routes/teams.js (hunk 8 above)
Comment (kelseyedmodo) [inline routes/teams.js]: Not quite. API serves this up as a totals object within the players array that has "three_pt_attmpts", "two_pt_attmps" inside. I've flattened it so there isn't a totals object and an average object.. but all those key/vals are in the main players array properly labeled.

File: routes/teams.js (hunk 6 above)
Comment (kelseyedmodo) [inline routes/teams.js]: It's not recognizing underscore here after I add up top.

var _ = require('underscore');

do I need to give it a more direct file path?

Comment (blakeparkinson) [inline routes/teams.js]: no thats good. u just need to merge master into your branch. Your branch is so far behind that it doesnt have alot of the latest stuff. u will have lots of merge conflicts to resolve tho.

Comment (kelseyedmodo) [inline routes/teams.js]: oh nice. thx.

Merge conflicts resolved and underscore added.

File: routes/teams.js [hunk 11]
```diff
@@ -1,105 +1,126 @@
...
+}
+
+
+function mongoInsertPlayers(team_id, team_document){
+  console.log("inserting into the DB");
+  db.open(function(err, db){
+    db.collection("players").update({team_id: team_id},
+    {$set: {team_id: team_document["team_id"], last_updated: new Date(), players: team_document["players"]}},
+    {upsert: true, multi:false}, function (err, upserted){
+      console.log("upsert complete");
```
Comment (blakeparkinson) [inline routes/teams.js]: id be more interested in outputting when an error occurs, i expect/hope that this would succeed.

**Comment by blakeparkinson** [comment]
omg :fire_engine:
